qmi/instruments/wl_photonics/wltf_n.py

```diff
@@ -0,0 +1,334 @@
+
+        """
+        response = self._read()
+        # Check that we got 'OK' as second item in response. The first is the echo of the command sent.
+        if response.split(self.response_terminator)[1] == "":
+            return
+
+        # Else try to check what went wrong
+        raise QMI_InstrumentException(f"Command {cmd} returned {response} instead of OK")
+
+    def _ask_int(self, cmd: str) -> int:
+        """Send a query and return an integer number response."""
+        self._write(cmd)
+        full_response = self._read().split("\n")  # First item should be the response, the second "OK".
+        try:
+            return int(search(r"[-\d.]+", full_response[0])[0])  # type: ignore
+        except ValueError as exc:
+            raise QMI_InstrumentException(f"Unexpected response to command {cmd!r}: {full_response!r}") from exc
+        except TypeError as exc:
+            raise QMI_InstrumentException(f"Unexpected response to command {cmd!r}: {full_response!r}") from exc
+
+    def _ask_float(self, cmd: str) -> float:
+        """Send a query and return a floating point response."""
+        self._write(cmd)
+        full_response = self._read().split("\n")  # First item should be the response, the second "OK".
+        if "Unknown" in full_response[0]:
+            # The wavelength has not been set or calibrated yet
+            return 0.0
+        try:
+            return float(search(r"[\d.]+", full_response[0])[0])  # type: ignore
+        except ValueError as exc:
+            raise QMI_InstrumentException(f"Unexpected response to command {cmd!r}: {full_response!r}") from exc
+        except TypeError as exc:
+            raise QMI_InstrumentException(f"Unexpected response to command {cmd!r}: {full_response!r}") from exc
+
+    @rpc_method
+    def open(self) -> None:
+        _logger.info("[%s] Opening connection to instrument", self._name)
+        self._check_is_closed()
+        self._transport.open()
+        super().open()
+        # Check the ID to update the ranges and log
+        _logger.info("Found instrument %s", self.get_idn())
+        # Check the current motor step position
+        self.get_motor_position()
+
+    @rpc_method
+    def close(self) -> None:
+        _logger.info("[%s] Closing connection to instrument", self._name)
+        self._check_is_open()
+        super().close()
+        self._transport.close()
+
+    @rpc_method
+    def get_idn(self) -> QMI_InstrumentIdentification:
+        """Read instrument info and return QMI_InstrumentIdentification instance. The instrument returns several
+        lines of data, example from the documentation:
+        'WL200: SN(201307374), MD(2018-11-23)\r\n',
+        'WL Range: 1021.509~1072.505nm(Step: 4654~556)\r\n',
+        'OK\r\n'.
+        This is processed to create the QMI_InstrumentIdentification object and also to update the wavelength and
+        step range class attributes.
+
+        Attributes:
```

Comment:
why is there an attributes section here?

Reply:
This function changes these class attributes/variables notably. Thus it would be good to mention this in the docstring and this way it gets highlighted well. I did not find any specific rule for this in PEP 256 - PEP 258, nor PEP 287, so I kind of made my own way. It seems to be in an gray area, so "at your own judgement" I guess.

Reply:
I see, for me it seems like this function itself has attributes based on the docstring

Reply:
Maybe I should add the `self.` in the names to make it more clear we talk about class attributes?

Reply:
Well I did put the `self.` there in the names, so it should be clear the attributes are class-wide.

Reply:
But I can rewrite it in another way.